### DISCUSSION TITLE
colorize (fill) a specific color how?

### DISCUSSION THREAD

**Discussion by brandnewcoat** [ImageMagick/ImageMagick#6388] [Help] [answered]
### ImageMagick version

7.1.1.9-1

### Operating system, version and so on

Manjaro

### Description

Hi there,

I want to learn how to colour a single coloured png image into another colour, other than just green, red, etc.

I have searched and haven't come close to colouring an image into lots of different coloured versions.  I can do this in GIMP using a colorize plugin but it takes a lot of time, which I would like to automate, once I have made the initial image.

My project is to make  lots of colours of each tile all at once without lots of clicks in GIMP and consuming time, as I would have my work cutout for me.

![@-black](https://github.com/ImageMagick/ImageMagick/assets/135526839/28c9aa3f-5318-4d1e-b113-a2d904857d16)![@-brown](https://github.com/ImageMagick/ImageMagick/assets/135526839/af37cf67-a01c-4298-902f-8d585143f01f)![@-green](https://github.com/ImageMagick/ImageMagick/assets/135526839/7ec8faab-287d-434e-b95c-dbf5204e7c62) x10
I first tried this, however I want to keep the alpha from being coloured and I want the smooth edges kept as well so it's just a copy with a different colour:
`convert -fill "#e03aec" -colorize 100 input.png output.png`

Is there a way of doing this?  I can't find an example which works and does what I want it to do.

Many thanks for reading and looking forward to your replies :-)

**Comment by fmw42** [comment, marked as answer]
Proper IM syntax is to read the input first, then the settings and operators, then the output.

Turn the alpha channel off before colorizing and then turn it back on.


![green](https://github.com/ImageMagick/ImageMagick/assets/13458218/a84f72ab-6131-4794-b60c-fb5c2d322410)

For IM 6,

`convert green.png -alpha off -fill "#e03aec" -colorize 100 -alpha on output.png
`

For IM 7

`magick green.png -alpha deactivate -fill "#e03aec" -colorize 100 -alpha activate output.png`

![output](https://github.com/ImageMagick/ImageMagick/assets/13458218/78504c93-e05f-4bb9-891d-95a2bb9ff974)

**Reply by brandnewcoat** [reply]
Thank you for your reply.  So I tried both methods and the result is still the same (result is just a solid pink background).  I'm wondering if there is something missing like an optional dependency or something not configured correctly;  how could I go about finding out?

**Reply by fmw42** [reply]
Post your IM 7 command and your result. And the input if not above.

**Reply by fmw42** [reply]
What do you get from 

`magick -version`

**Reply by brandnewcoat** [reply]
Thank you for being patient.  It was  a fault on my part, because the thumbnail was displaying as a solid colour so I assumed that it must be so.  When I uploaded it to show you the result it displayed correctly, and then corrected the thumbnail when I renamed it.

Thank again for your help :-)
